resources/main.mjs

```diff
@@ -260,12 +265,59 @@ class MainBenchmarkClient {
         return JSON.stringify(this._measuredValuesList, undefined, indent);
     }
 
+    _getFormattedCSVResult() {
+        let tests = [];
+        // The CSV format is similar to the details view table. Each measurement is a row with
+        // the name and N columns with the measurement for each iteration:
+        // ```
+        // Measurement,#1,...,#N
+        // TodoMVC-JavaScript-ES5/Total,num,...,num
+        // TodoMVC-JavaScript-ES5/Adding100Items,num,...,num
+        // TodoMVC-JavaScript-ES5/Adding100Items/Sync,num,...,num
+        // TodoMVC-JavaScript-ES5/Adding100Items/Async,num,...,num
+        // ...
+        // TodoMVC-JavaScript-ES6/Total,num,...,num
+        // TodoMVC-JavaScript-ES6/Adding100Items,num,...,num
+        // TodoMVC-JavaScript-ES6/Adding100Items/Sync,num,...,num
+        // TodoMVC-JavaScript-ES6/Adding100Items/Async,num,...,num
+        // ```
+        const firstIterationTests = this._measuredValuesList[0].tests;
+        for (const suiteName in firstIterationTests) {
+            tests.push([`${suiteName}/Total`]);
+            for (const testName in firstIterationTests[suiteName].tests) {
+                tests.push([`${suiteName}/${testName}`]);
+                for (const subtestName in firstIterationTests[suiteName].tests[testName].tests)
+                    tests.push([`${suiteName}/${testName}/${subtestName}`]);
+            }
+        }
+
+        // Now push each iteration onto the end of the array
+        for (const measuredValue of this._measuredValuesList) {
+            let index = 0;
+            for (const suiteName in measuredValue.tests) {
+                const suiteResults = measuredValue.tests[suiteName];
+                tests[index++].push(suiteResults.total);
+                for (const testName in suiteResults.tests) {
+                    tests[index++].push(suiteResults.tests[testName].total);
+                    for (const subtestName in suiteResults.tests[testName].tests)
+                        tests[index++].push(suiteResults.tests[testName].tests[subtestName]);
+                }
+            }
+        }
+
+        const csv = [["Name"].concat(this._measuredValuesList.map((_, i) => `#${i + 1}`)).join(",")];
+        for (const test of tests)
+            csv.push(test.join(","));
```

Comment:
Sorry, I forgot to press send on the reviews... (apparently I've never done that :D).

You could simplify this completely by using the `Metric` objects which would make it consistent with the rest of the system (I've deliberately not used that for the JSON to keep it backwards compatible).

You can replace all the code here with just a single line:
```
const csv = this.metrics.map(metric => [metric.name, ...metric.values].join(","))
```